great_expectations/expectations/metrics/multicolumn_map_metrics/multicolumn_sum_equal.py

```diff
@@ -10,9 +14,20 @@ class MulticolumnSumEqual(MulticolumnMapMetricProvider):
     condition_value_keys = ("sum_total",)
     condition_domain_keys = ("batch_id", "table", "column_list")
 
-    # TODO: <Alex>ALEX -- temporarily only a Pandas implementation is provided (others to follow).</Alex>
+    # TODO: <Alex>ALEX -- temporarily only Pandas and SQLAlchemy implementations are provided (others to follow).</Alex>
     @multicolumn_condition_partial(engine=PandasExecutionEngine)
     def _pandas(cls, column_list, **kwargs):
         sum_total = kwargs.get("sum_total")
         row_wise_cond = column_list.sum(axis=1) == sum_total
         return row_wise_cond
+
+    @multicolumn_condition_partial(engine=SqlAlchemyExecutionEngine)
+    def _sqlalchemy(cls, column_clause, **kwargs):
+        sum_total = kwargs.get("sum_total")
+        engine = kwargs.get("_sqlalchemy_engine")
+        table = kwargs.get("_table")
+        query = sa.select(column_clause).select_from(table)
+        with engine.connect() as connection:
```

Comment:
@NathanFarmer In our architecture, the actual "execute" operation for SQL and Spark happens in those metric provider helper functions (e.g., `_sqlalchemy_map_condition_rows()` and others), rather than in the methods in the metric implementation class.  Hence, when the decorator of such a method is a "condition_partial" type, then other than for Pandas, the metric implementation (the one in the metric class for the three execution engines) only computes the condition -- as a query.  Then, the helper function constructs the query based on this condition query (e.g., `unexpected_condition`) and the` selectable` returned when getting the "compute_domain" from the execution engine.  This way they actual computation is deferred (again, only for SQL and Spark engines).  Happy to discuss next week.  Thanks.

Reply:
I have made these logic changes and will be including them in the new PR.